doc/src/manual/constructors.md

```diff
@@ -114,9 +114,11 @@ violate this invariant. Of course, messing around with an object's internals uni
 You (or someone else) can also provide additional outer constructor methods at any later point, but
 once a type is declared, there is no way to add more inner constructor methods. Since outer constructor
 methods can only create objects by calling other constructor methods, ultimately, some inner constructor
-must be called to create an object. This guarantees that all objects of the declared type must come into
-existence by a call to one of the inner constructor methods provided with the type, thereby giving
-some degree of enforcement of a type's invariants.
+must be called to create an object. This means that usually at any given moment all existing objects of
+the declared type have come into existence by a call to one of the inner constructor methods provided
+with the type, thereby giving some degree of enforcement of a type's invariants. An exception to this
+rule are objects constructed using [Incomplete Initialization](@ref), which may bypass the inner
+constructor methods of a type.
```

Comment:
This isn't really true: an inner constructor can perform incomplete initialization, but you aren't bypassing an inner constructor by doing this.

Reply:
Would you not describe this as the inner constructor of `S` being bypassed by the incomplete initialization (in the `T` constructor):
```
julia> struct S
         v::Int
         function S(v::Int)
           (v ∈ (700, 703)) || error("invalid value")
           new(v)
         end
       end

julia> struct T{U}
         v::U
         T{U}() where {U <: Any} = new{U}()
       end

julia> T{S}()
T{S}(S(0))

julia> S(0)
ERROR: invalid value
```

Reply:
> Well the same would apply to construction via `reinterpret`, or `Array{S}(undef, 1)`, but you're not really "constructing" it at that point.

I'm fine with calling it, e.g., "creation", instead of "construction"; but I think there should be some kind of warning about these situations in the manual. Currently the manual says something which is both obviously false and potentially dangerously misleading:

> This guarantees that all objects of the declared type must come into existence by a call to one of the inner constructor methods provided with the type